src/codec/MapAddEntryListenerCodec.ts

```diff
@@ -0,0 +1,82 @@
+/* tslint:disable */
+import ClientMessage = require('../ClientMessage');
+import {BitsUtil} from '../BitsUtil';
+import Address = require('../Address');
+import {Data} from '../serialization/Data';
+import {MapMessageType} from './MapMessageType';
+
+var REQUEST_TYPE = MapMessageType.MAP_ADDENTRYLISTENER;
+var RESPONSE_TYPE = 104;
+var RETRYABLE = false;
+
+
+export class MapAddEntryListenerCodec {
+
+
+    static calculateSize(name:string, includeValue:boolean, listenerFlags:number, localOnly:boolean) {
+        // Calculates the request payload size
+        var dataSize:number = 0;
+        dataSize += BitsUtil.calculateSizeString(name);
+        dataSize += BitsUtil.BOOLEAN_SIZE_IN_BYTES;
+        dataSize += BitsUtil.INT_SIZE_IN_BYTES;
+        dataSize += BitsUtil.BOOLEAN_SIZE_IN_BYTES;
+        return dataSize;
+    }
+
+    static encodeRequest(name:string, includeValue:boolean, listenerFlags:number, localOnly:boolean) {
+        // Encode request into clientMessage
+        var clientMessage = ClientMessage.newClientMessage(this.calculateSize(name, includeValue, listenerFlags, localOnly));
+        clientMessage.setMessageType(REQUEST_TYPE);
+        clientMessage.setRetryable(RETRYABLE);
+        clientMessage.appendString(name);
+        clientMessage.appendBoolean(includeValue);
+        clientMessage.appendInt32(listenerFlags);
+        clientMessage.appendBoolean(localOnly);
+        clientMessage.updateFrameLength();
+        return clientMessage;
+    }
+
+    static decodeResponse(clientMessage:ClientMessage, toObjectFunction:(data:Data) => any) {
```

Comment:
not all messages have data to deserialize. but the function signature should be same